### PR TITLE
Move memory from utils to training

### PR DESCRIPTION
**Pull request by RdoubleA**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Addresses #1437.

#### Changelog
What are the changes made in this PR?

#### Test plan
Please make sure to do each of the following if applicable to your PR. (If you're not sure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.)

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Example of docstring: https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285
Example in our docs: https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models

- [ ] I did not change any public API;
- [ ] I have added an example to docs or docstrings;

### CODE REVIEW

File: torchtune/utils/__init__.py [hunk 1]
```diff
@@ -51,9 +41,15 @@
...
-    "set_activation_checkpointing",
     "set_seed",
     "torch_version_ge",
+    "DEFAULT_PROFILE_DIR",
+    "DEFAULT_PROFILER_ACTIVITIES",
+    "DEFAULT_SCHEDULE",
+    "DEFAULT_TRACE_OPTS",
+    "DummyProfiler",
+    "PROFILER_KEY",
+    "setup_torch_profiler",
```
Comment (felipemello1) [inline torchtune/utils/__init__.py:52]: do you mind double checking those? i dont understand what is being added here or why

**Comment by felipemello1** [comment]
there are conflicts. I can approve after that.

File: torchtune/utils/__init__.py (hunk 1 above)
Comment (RdoubleA) [reply torchtune/utils/__init__.py:49]: ah good catch, these should be removed

File: docs/source/api_ref_training.rst [hunk 2]
```diff
@@ -38,6 +38,21 @@ Utilities for working in a reduced precision setting.
     validate_expected_param_dtype
     get_quantizer_mode
 
+.. _ac_label:
+
+Memory Management
```
Comment (ebsmothers) [inline docs/source/api_ref_training.rst:43]: This is a duplicate of the section immediately following

**Review by ebsmothers — APPROVED** [review]
One comment, other than that looks good

File: torchtune/utils/__init__.py [hunk 3]
```diff
@@ -26,29 +26,17 @@
...
 from ._version import torch_version_ge
 from .logging import get_logger
-from .memory import (  # noqa
-    cleanup_before_training,
-    create_optim_in_bwd_wrapper,
-    get_memory_stats,
-    log_memory_stats,
-    OptimizerInBackwardWrapper,
-    register_optim_in_bwd_hooks,
-    set_activation_checkpointing,
```
Comment (felipemello1) [inline torchtune/utils/__init__.py:36]: some of the deleted imports are still in __ all __

**Review by felipemello1 — CHANGES_REQUESTED** [review]
some o

**Review by felipemello1 — APPROVED** [review]
